tfhe/src/core_crypto/gpu/slice.rs

```diff
@@ -0,0 +1,194 @@
+            ptr,
+            _len: len,
+            _phantom_1: PhantomData,
+            _phantom_2: PhantomData,
+        }
+    }
+
+    /// # Safety
+    pub(crate) unsafe fn as_c_ptr(&self) -> *const c_void {
+        self.ptr.as_c_ptr()
+    }
+}
+
+impl<'a, T> CudaSliceMut<'a, T>
+where
+    T: Numeric,
+{
+    pub(crate) unsafe fn new(ptr: CudaPtr, len: usize) -> CudaSliceMut<'a, T> {
+        CudaSliceMut {
+            ptr,
+            len,
+            _phantom_1: PhantomData,
+            _phantom_2: PhantomData,
+        }
+    }
+
+    /// # Safety
+    pub(crate) unsafe fn as_mut_c_ptr(&mut self) -> *mut c_void {
+        self.ptr.as_mut_c_ptr()
+    }
+
+    /// # Safety
+    pub(crate) unsafe fn as_c_ptr(&self) -> *const c_void {
+        self.ptr.as_c_ptr()
+    }
+
+    /// Copies data between two `CudaSlice`
+    ///
+    /// # Safety
+    ///
+    /// - [CudaStream::synchronize] __must__ be called after the copy
+    /// as soon as synchronization is required
+    pub unsafe fn copy_from_gpu_async(&mut self, src: &Self, stream: &CudaStream)
+    where
+        T: Numeric,
+    {
+        assert!(self.len() >= src.len());
+        let size = src.len() * std::mem::size_of::<T>();
+        // We check that src is not empty to avoid invalid pointers
+        if size > 0 {
+            cuda_memcpy_async_gpu_to_gpu(
+                self.as_mut_c_ptr(),
+                src.as_c_ptr(),
+                size as u64,
+                stream.as_c_ptr(),
+            );
+        }
+    }
+
+    #[allow(dead_code)]
+    /// Copies data of a `CudaSlice` to the CPU
+    ///
+    /// # Safety
+    ///
+    /// - [CudaStream::synchronize] __must__ be called after the copy
+    /// as soon as synchronization is required
+    pub unsafe fn copy_to_cpu_async(&self, dest: &mut [T], stream: &CudaStream)
+    where
+        T: Numeric,
+    {
+        assert!(self.len() <= dest.len());
+        let size = self.len() * std::mem::size_of::<T>();
+        // We check that src is not empty to avoid invalid pointers
+        if size > 0 {
+            cuda_memcpy_async_to_cpu(
+                dest.as_mut_ptr().cast::<c_void>(),
+                self.as_c_ptr(),
+                size as u64,
+                stream.as_c_ptr(),
+            );
+        }
+    }
+
+    /// Returns the number of elements in the vector, also referred to as its ‘length’.
+    pub fn len(&self) -> usize {
+        self.len
+    }
+
+    pub(crate) fn get_mut<R>(&mut self, range: R) -> Option<CudaSliceMut<T>>
+    where
+        R: std::ops::RangeBounds<usize>,
+        T: Numeric,
+    {
+        let (start, end) = range_bounds_to_start_end(self.len(), range).into_inner();
+
+        // Check the range is compatible with the vec
+        if end <= start || end > self.len - 1 {
+            None
+        } else {
+            // Shift ptr
+            let shifted_ptr: *mut c_void = unsafe {
+                self.ptr
+                    .as_mut_c_ptr()
```

Comment:
Minor thing but we could use [byte_add](https://doc.rust-lang.org/core/primitive.pointer.html#method.byte_add)
`self.ptr.byte_add(start * std::mem::sizeof::<T>())`

Reply:
I get this compilation error:
```
error: current MSRV (Minimum Supported Rust Version) is `1.73.0` but this item is stable since `1.75.0` 
```
Do you know how I could fix it?

Reply:
Only way to fix is to bump the minimum rust version that is in the tfhe/Cargo.toml

But that has some implications for users

Reply:
for 0.7 I think it's fine yes, for 0.6 it would be breaking I'm afraid

Reply:
Ok so what I can do is not use `byte_add` in this PR, then I'll do another small PR just to increase the MSRV and use byte_add there. Wdyt?

Reply:
yep if this PR targets 0.6 that sounds good